### PR TITLE
flawlessapp: new cask

### PR DESCRIPTION
**Pull request by yurikoles**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/flawlessapp.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'flawlessapp' do
+  version :latest
+  sha256 :no_check
+
+  # dl.devmate.com/com.flawless.app.FlawlessMac was verified as official when first introduced to the cask
+  url 'https://dl.devmate.com/com.flawless.app.FlawlessMac/FlawlessMac.zip'
+  name 'Flawless App'
```
Comment (commitay) [inline Casks/flawlessapp.rb]: Please add the appcast and version / sha256

http://updates.devmate.com/com.flawless.app.FlawlessMac.xml

**Review by commitay — CHANGES_REQUESTED** [review]

**Review by commitay — APPROVED** [review]

**Comment by vitorgalvao** [comment]
I don’t see a way to access the download without giving them an email.

**Comment by commitay** [comment]
https://flawlessapp.io/download is public but you can't get there from the `homepage`.

@vitorgalvao Do you want to refuse this as a walled build?

**Comment by claui** [comment]
(Just chiming in even though I’m not addressed.) My two cents: Whenever I fail to find a decent download link on a page, I find myself inclined to browse to `/download` manually.

That alone would lead me to say that URL is public, no matter if the homepage links to it or not.

**Comment by vitorgalvao** [comment]
> (Just chiming in even though I’m not addressed.)

You’re always welcome to chime in! As is any maintainer and most users!

> Whenever I fail to find a decent download link on a page, I find myself inclined to browse to `/download` manually.

That’s a valid point and I’d be inclined to agree *if* there was no clue on the page at all as to how to download.

*However*, the homepage does tell you how to download. But it also tells you that you *need*<sup>1</sup> to give them your email to proceed to the download. Yes, for the moment you can just give them any fake email, but I don’t expect that that will be most people’s first reaction.

The fact that the download page is behind a wall (you need to give them an email) classifies it in my mind as walled. Even though we can manually get to the `/download` page, that is us *circumventing their desired system*. If we ask ourselves “what did the developers intend this to be”, I’d say the answer is “walled”.

Also, apologies for the late reply @commitay. It fell through the cracks and I had forgotten about it.

---

1. If you click “Download” without entering an email, it’ll not let you proceed.

**Comment by claui** [comment]
> If we ask ourselves “what did the developers intend this to be”, I’d say the answer is “walled”.

Fair point.

Thanks @vitorgalvao for elaborating your rationale, and of course for your welcoming words 😊  

I think I’ve reconsidered a bit my notion of where an offer actually starts to be walled. From that point of view, I agree in that we should rather not accept `flawlessapp` into the core tap.

**Comment by commitay** [comment]
> Also, apologies for the late reply @commitay. It fell through the cracks and I had forgotten about it.

No problem, I had forgotten as well!
____

Passing on this as a walled build, please fell free to resubmit it if upstream makes the download available without requiring an email.

**Pull request closed without merging**